### PR TITLE
Sites UI

### PR DESCRIPTION
**Pull request by jamesmorgan**
* [x] improved CSS 
* [x] `EnititMetaData` constant
* [x] mocking up on dash and sites
* [x] `SiteMoodStatus` directive

@andygray I fixed the issue. It was tricky to find, I ended up debugging ui-router which was swallowing an exception as part of a resolver. 

Error:
*"[$injector:unpr] Unknown provider: SiteServiceProvider <- SiteService
http://errors.angularjs.org/1.3.8/$injector/unpr?p0=SiteServiceProvider%20%3C-%20SiteService"*

screenshot 2015-01-25 17 04 29

### CODE REVIEW

File: app/js/ui-routes.js [hunk 1]
```diff
@@ -48,7 +53,25 @@
...
+                    /**
+                     * @ngInject
+                     */
+                    sites: function (SitesService, SiteMoodService, _) {
+
+                        return SitesService.list()
+                            .then(function (sites) {
+                                return sites.map(function (site) {
+                                    SiteMoodService.getForSite(site).then(function (siteMood) {
+                                        site['mood'] = siteMood; // TODO Q: use extend?
```
Comment (jamesmorgan) [inline app/js/ui-routes.js]: I see where you are going with this, its should work when its read only views. The problem is that once you bolt on the `mood` object you can no longer just save this without removing the property once again without it being pushed to `firebase`. This isn't problem though and I think for read only views it should work really well. :+1:

File: app/js/simpleLogin.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ angular.module('simpleLogin', ['firebase', 'firebase.utils', 'changeEmail'])
                 user: null,
 
                 getUser: function () {
-                    return auth.$waitForAuth();
+                    return auth.$waitForAuth().then();
```
Comment (jamesmorgan) [inline app/js/simpleLogin.js]: What is this for?

Comment (andygray) [inline app/js/simpleLogin.js]: Was thinking of loading the user information and extending the user object

File: app/css/AdminLTE_Custom.css [hunk 3]
```diff
@@ -21,6 +21,10 @@
     font-size: 25px;
 }
 
+.icon-button {
```
Comment (andygray) [inline app/css/AdminLTE_Custom.css]: @jamesmorgan try and avoid using entity names in css classes like `site-mood-button` as when we reuse it doesn't make sense...

File: app/js/entity/site-mood/directives/SiteMoodStatus.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            restrict: 'E',
+            replace: true,
+            scope: {
+                site: '='
+            },
+            templateUrl: 'partials/templates/site-mood-status.tpl.html',
+            link: function ($scope) {
+
+                // TODO we need a way of injecting users into directives
+                var currentUser = null;
```
Comment (andygray) [inline app/js/entity/site-mood/directives/SiteMoodStatus.js]: this has been lift and shifted...

File: app/css/AdminLTE_Custom.css (hunk 3 above)
Comment (jamesmorgan) [inline app/css/AdminLTE_Custom.css]: :+1:

File: app/index.html [hunk 5]
```diff
@@ -180,14 +185,20 @@
...
+<<<<<<< HEAD
+<script src="js/dashboard/site/_module.js"></script>
+<script src="js/dashboard/site/SiteBoxCtrl.js"></script>
+=======
 <script src="js/sidebar/_module.js"></script>
 <script src="js/sidebar/SidebarCtrl.js"></script>
 
 <script src="js/navbar/_module.js"></script>
 <script src="js/navbar/ProfileCtrl.js"></script>
+>>>>>>> master
```
Comment (jamesmorgan) [inline app/index.html]: Merge conflicts!

File: app/index.html [hunk 6]
```diff
@@ -180,14 +185,20 @@
 <script src="js/dashboard/organisation/_module.js"></script>
 <script src="js/dashboard/organisation/OrganisationBoxCtrl.js"></script>
 
+<<<<<<< HEAD
```
Comment (jamesmorgan) [inline app/index.html]: Merge conflicts!

File: app/index.html [hunk 7]
```diff
@@ -180,14 +185,20 @@
...
+<script src="js/dashboard/site/_module.js"></script>
+<script src="js/dashboard/site/SiteBoxCtrl.js"></script>
+=======
```
Comment (jamesmorgan) [inline app/index.html]: Merge conflicts!

File: app/js/dashboard/DashboardCtrl.js [hunk 8]
```diff
@@ -4,10 +4,16 @@
...
      */
-    function DashboardCtrl($log, user, $state) {
+    function DashboardCtrl($log, user, $state, organisations, sites, EntityMetaData) {
         var vm = this;
         vm.loggedIn = user;
         vm.breadcrumbs = $state.current.data.breadcrumbs;
+        vm.metaData = EntityMetaData;
+
+        // quick and nasty!
+        vm.organisationsCount = organisations && organisations.length;
```
Comment (jamesmorgan) [inline app/js/dashboard/DashboardCtrl.js]: I think you can do a `$count()` on a collection `$ref` - we can fix in the future!

File: app/js/ui-routes.js [hunk 9]
```diff
@@ -1,19 +1,29 @@
 (function () {
     'use strict';
 
+    // Q: how do we inject EntityMetaData in so we can use that constant class?
     var Breadcrumbs = {
```
Comment (jamesmorgan) [inline app/js/ui-routes.js]: I will fix this in my next PR.

**Comment by jamesmorgan** [comment]
You should rebase and then get this in, can you do it tonight?

**Comment by jamesmorgan** [comment]
Sites is looking really good :+1:

**Pull request merged**